src/main/java/net/pms/util/ImagesUtil.java

```diff
@@ -92,57 +158,1394 @@ public static void auto(InputStream input, OutputStream output, int exifOrientat
+	 * padding. Preserves aspect ratio. Format support is limited to
+	 * that of {@link ImageIO}.
+	 *
+	 * @param inputStream the source image in a supported format.
+	 * @param width the new width.
+	 * @param height the new height.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param updateMetadata whether or not new metadata should be updated
+	 *                       after image transformation. This should only be
+	 *                       disabled if the output image won't be kept/reused.
+	 * @param dlnaCompliant whether or not the output image should be
+	 *                      restricted to DLNA compliance. This also means that
+	 *                      the output can be safely cast to {@link DLNAImage}.
+	 * @param dlnaThumbnail whether or not the output image should be
+	 *                      restricted to DLNA thumbnail compliance. This also
+	 *                      means that the output can be safely cast to
+	 *                      {@link DLNAThumbnail}.
+	 * @param padToSize Whether padding should be used if source aspect doesn't
+	 *                  match target aspect.
+	 * @return The scaled, {@code null} if the source is {@code null} or the
+	 *         source image if the operation fails.
+	 * @throws IOException if the operation fails.
+	 */
+	public Image scaleImage(
+		InputStream inputStream,
+		int width,
+		int height,
+		ScaleType scaleType,
+		boolean updateMetadata,
+		boolean dlnaCompliant,
+		boolean dlnaThumbnail,
+		boolean padToSize
+	) throws IOException {
+		return transcodeImage(inputStream, width, height, scaleType, ImageFormat.SOURCE, updateMetadata, dlnaCompliant, dlnaThumbnail, padToSize);
+	}
+
+	/**
+	 * Scales an image to the given dimensions. Scaling can be with or without
+	 * padding. Preserves aspect ratio. Format support is limited to
+	 * that of {@link ImageIO}.
+	 *
+	 * @param inputByteArray the source image in a supported format.
+	 * @param width the new width.
+	 * @param height the new height.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param updateMetadata whether or not new metadata should be updated
+	 *                       after image transformation. This should only be
+	 *                       disabled if the output image won't be kept/reused.
+	 * @param dlnaCompliant whether or not the output image should be
+	 *                      restricted to DLNA compliance. This also means that
+	 *                      the output can be safely cast to {@link DLNAImage}.
+	 * @param dlnaThumbnail whether or not the output image should be
+	 *                      restricted to DLNA thumbnail compliance. This also
+	 *                      means that the output can be safely cast to
+	 *                      {@link DLNAThumbnail}.
+	 * @param padToSize Whether padding should be used if source aspect doesn't
+	 *                  match target aspect.
+	 * @return The scaled, {@code null} if the source is {@code null}.
```

Comment:
@Sami32 Actually I changed that for all of them, but had forgotten to update all of the Javadocs (there has been a LOT of back and forth with these methods, so keeping the Javadocs updates has been a challenge). Neither returns the source image if the operation fails, they throw an ```Exception``` instead letting the caller decide how to handle it.